_posts/2022-04-04-valheim-server.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: Hosting a Valheim Server on AWS Part:1
+date: 2022-04-04
+tags: code ecs fargate cloudnative valheim deployment dedicated-server devops gaming aws
+author: Joseph Miocevich
+image: img/valheim-logo.webp
+description:
+  How to set up a dedicated server for Valheim on AWS, with some discord notifications
+---
+
+![Header](/img/valheim-logo.webp)
+
+Do you want to set up your own Valheim Server? Are you sick of waiting for your friend to figure out how to port forward their local server? Has your wallet been raided by an 'easy' online game hosting solution? Then this is the blog post for you! Read on if you want to become a certified Viking nerd.
+
+Now, if you are like me, where I decide 9 pm is when I should do that documentation I've been procrastinating on, the hours that my friends play and I play rarely aligns, hence the need for a dedicated server. I’ve seen a few online tutorials on how to set up your own Valheim Server, but I took it as an opportunity to do what any semi self-loathing programmer does.... ignore it all and decide I can do better...I mean attempt to do it myself as a learning opportunity.
+
+If you just want the code, here you go: [ Valheim-Server-on-AWS-Part-1 ](https://github.com/JMiocevich/Valheim-Server-on-AWS-Part-1)
+
+We are gonna separate this post into two parts. Part 1 will be the bare minimum to get a server up and running, whilst part 2 will include some fancier bells and whistles.
+
+So let's cover some basic criteria on what creating your own virtual Vikings adventure land will entail:
+
+- Infrastructure as code using AWS SAM (minimal clicksops)
+- Easy deployment for the technically challenged
+- Secure and resilient architecture
+- Cost competitive against out of the box services
+- Cheap
+- Ability to Auto stop server when game world is not in use
+- Get Notifications on the game state in discord (server start/stop/when a player joins)
```

Comment:
Do you plan on discussing this (here or in a subsequent post?)

Picking up interesting pieces of the architecture and discussing that in more detail is good - this is a technical blog.

Reply:
Re-reading the section above - I think you intend to do that in a future post.  

If it confused me, it will likely confuse other readers.

Suggest that you clearly split this into three sections:
* The overall requirements (this list?)
* The requirements you cover in this post
* The requirements you will cover in a future post (enough to keep them wanting more)

Reply:
Again, to cover the technical content for the blog, an architecture diagram would be useful. 

Either here, or under the discussion section

Reply:
Will draw up a architecture diagram, and clarify which posts will cover content